### ISSUE TITLE
Returning correct bin centers

### ISSUE THREAD

**Issue by sukhdeep1989** [rmjarvis/TreeCorr#112]
Currently Treecorr returns either the geometric bin center or the pair weighted bin center. They both tend to be biased as there is weighting with the signal also. See appendix D of https://arxiv.org/pdf/1811.06499.pdf for discussion.

Is it possible to return correct bin centers in treecorr? Perhaps letting user input a theoretical signal to be used along with pair weights. 

If you think this kind of calculation is outside the scope of treecorr, then maybe add a warning to the user that they should pay a careful attention to this issue? This is mostly a problem when theory is computed at the wrong bin centers instead of being properly binned as well.
Replacing current meanr definition with the simple 0.5*(r_min+r_max) will make it less biased for the 2D cosmological correlation functions. This is because most of our signals scale as ~1/r and pair weights scale as r.

**Comment by rmjarvis** [comment]
There is also the pair-weighted mean(logr), which I believe is normally the best one to use for things like this, since the signal is generally locally approximately a power law.  So exp(meanlogr) is normally what people use I think.  Not sure if you looked at that option, but it doesn't seem to be mentioned in your paper.

And, you can certainly use your suggested 0.5*(r_min+r_max) if you want.  From a TreeCorr correlation object, the code would be
```
r_alt = 0.5 * (corr.left_edges + corr.right_edges)
```
It looks like I might not have added these attributes to the documentation, but they have been around for a couple versions now.

I'm not so keen on adding a user-supplied weight function, since this would need to be called from C++, and python call backs are not particularly efficient.  I think if the differences among these various options have a non-negligible effect for your science case, the best bet is to use smaller bins where the differences among them are smaller.

**Comment by sukhdeep1989** [comment]
Sorry, I hadn't looked at the pair weighted mean(logr). By my calculation the error in mean(logr) should be second order in log(bin size), assuming signal is 1/r, which is probably ok for most applications. 

To clarify my thinking, I normally bin the theory with pair weight as well, so is not big issue for me. It has come in discussions with people a few times and I thought it will be helpful if Treecorr clarifies this issue and allows one to compute the right center. Using smaller bins is not always an option due to covariance issues, etc. I wasn't advocating for a user supplied functions, but rather user supplied (r,xi) values, which can then be used internally in treecorr to construct an interpolation object and use it in conjunction with pair weights.

I would be satisfied with mean(logr), so we can close the issue.

**Comment by rmjarvis** [comment]
As a historical note, exp(meanlogr) and r_nom used to be the only ones I output, since I though the former was normally the right choice if you were comparing to theory.  But some users asked me to calculate the pair-weighted meanr as well, so that's now there too.  Probably if it hadn't been there, you might not have been tempted to use that over meanlogr.  Sorry for that.

I agree that I should add some docs pointing out the nuances about this would be useful.  To make this maximally effective, where in the docs would **you** probably have looked for this information when you were getting going on this?  The tutorial?  One of the doc strings?  Elsewhere in the Sphinx docs?

**Comment by sukhdeep1989** [comment]
I had looked at the attributes of two point correlations, e.g. https://rmjarvis.github.io/TreeCorr/_build/html/nn.html to see what different attributes are available. For some reason I didn't pay close attention to mean logr and missed it, Sorry.
Another place I looked at was the description on binning (https://rmjarvis.github.io/TreeCorr/_build/html/binning.html). I would recommend adding a small section on bin centering here.

**Comment by rmjarvis** [comment]
Good. Thanks.  I think the binning page would be a good spot for it.  Cheers!

**Comment by rmjarvis** [comment]
I finally got around to doing this.  I added an "Output Quantities" to the Binning page:

https://rmjarvis.github.io/TreeCorr/_build/html/binning.html#output-quantities

The last two paragraphs of this section are most relevant to the points you raised here.  Let me know if there are any critical points you think are missing.  I do point readers to your paper for more information, but I'm happy to add more here if you think it would be helpful.
